### DISCUSSION TITLE
2.1.1rc2

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#24788] [General]
The release candidate:
https://dist.apache.org/repos/dist/dev/superset/2.1.1rc2/

Git tag for the release:
https://github.com/apache/superset/tree/2.1.1rc2

The Change Log for the release:
https://github.com/apache/superset/blob/2.1.1rc2/CHANGELOG.md

The Updating instructions for the release:
https://github.com/apache/superset/blob/2.1.1rc2/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

**Comment by eschutho** [comment]
Please add any testing results that you find in the discussion below.

**Comment by sfirke** [comment]
I'm experiencing a regression that is a little hard to pin down.  I copied my Superset metadata DB and created a new instance on the copy using the 2.1.1rc2 image.

A dashboard native filter is failing to load its first value by default.  It's set up like this:
![image](https://github.com/apache/superset/assets/7569808/6d77f71c-9f4b-4770-94da-5f4efef7cad4)

It fails to load any value:
![image](https://github.com/apache/superset/assets/7569808/d3173eaf-c4fa-4896-9463-e15730a83f4e)

This causes the dashboard to fail to load, with a spinning Superset logo.  If I refresh the dashboard, it will now load, but the charts using that filter are broken because that value is referenced in a Jinja virtual dataset.  Everything works if I specify a filter value manually, or change the native filter to use a specific value as default.

This behavior worked in 2.1.0.

**UPDATE**: I seem to have replicated this bug in the 2.1.1rc2 ephemeral environment, feel free to log in and take a look.    And the bug is not present in 3.0.0rc2; there my existing dashboard works correctly.

**Reply by sfirke** [reply]
I might have gotten it replicated in the 2.1.1rc2 ephemeral, without Jinja.  See [this dashboard](http://34.219.107.205:8080/superset/dashboard/11/), you might need to log in first:

![image](https://github.com/apache/superset/assets/7569808/c2d70dcd-dbd8-4ffa-bfec-d03aea2339ff)

Let me know if there's a reason that filter config actually should be failing to select the first value by default?

**Reply by eschutho** [reply]
Thanks @sfirke! It looks like the vote for rc2 didn't pass, so we'll see if we can get a fix for this or revert the PR for rc3.

**Reply by eschutho** [reply]
Hi @sfirke, I git bisected with your dashboard (thank you!) and found the missing PR, so I cherried it in and built a new ephemeral. It seems to be working, if you want to try it out: http://54.202.87.4:8080/superset/dashboard/11/?native_filters_key=ALp-Gi1w5p7plsXY_Djux9tCxT1l6FluyeRKGFVNnBkjVJ9lWb0eAi90cInTXBhn

**Reply by sfirke** [reply]
Yes this did it!  I applied the docker hub image [pr-24224-dev](https://hub.docker.com/layers/apache/superset/pr-24224-dev/images/sha256-b1201429565f99cb682303550528ea08c69e56ac01d32bbefbd44d217c159165?context=explore) to my test environment and that fixed this bug 🙌

**Comment by sfirke** [comment]
Awesome! I wonder if we can easily generate a docker image that I can plug
into my own test environment to test there. I think if you create a "don't
merge this" dummy PR it will build an image and send to Docker hub?

**Reply by eschutho** [reply]
Yeah, I think you can use this PR for the 2.1.1 release: https://github.com/apache/superset/pull/24224

**Comment by michael-s-molina** [comment]
[2.1.1 RC3 discussion](https://github.com/apache/superset/discussions/25019) is out.
